### PR TITLE
Fix Assertion failed 'ins == INS_add' for madd/msub

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/pull/61037#issuecomment-959156035 - it didn't expect INS_adds there (GT_ADD can set flags). Also, I decided to rewrite it to use a separate GT_MADD op (see https://github.com/dotnet/runtime/pull/61045#discussion_r741534443)

cc @jakobbotsch

### CODE REVIEW

File: src/tests/JIT/Regression/JitBlue/Runtime_61037/Runtime_61037.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// Hits JIT assert in Release:
+// Assertion failed 'ins == INS_add' in 'Program:Main(Fuzzlyn.ExecutionServer.IRuntime)' during 'Generate code' (IL size 28)
+// 
+//     File: D:\a\_work\3\s\src\coreclr\jit\emitarm64.cpp Line: 13602
+// 
+public class C0
+{
+}
+
+public class Program
```
Comment (jakobbotsch) [inline src/tests/JIT/Regression/JitBlue/Runtime_61037/Runtime_61037.cs:14]: Rename to Runtime_61037? Also missing license header at the top.

File: src/coreclr/jit/codegenarm64.cpp [hunk 2]
```diff
@@ -9564,4 +9564,47 @@ void CodeGen::instGen_MemoryBarrier(BarrierKind barrierKind)
...
+//
+// Arguments:
+//     tree - GT_MADD tree where op1 or op2 is GT_ADD
+//
+void CodeGen::genCodeForMadd(GenTreeOp* tree)
+{
+    assert(tree->OperIs(GT_MADD) && varTypeIsIntegral(tree) && !(tree->gtFlags & GTF_SET_FLAGS));
+    genConsumeOperands(tree);
+
+    GenTree *a, *b, *c;
```
Comment (SingleAccretion) [inline src/coreclr/jit/codegenarm64.cpp:9578]: Nit: our style guide forbids such one-line declarations.
```suggestion
    GenTree* a;
    GenTree* b;
    GenTree* c;
```

**Review by jakobbotsch — APPROVED** [review]
LGTM

**Comment by EgorBo** [comment]
Failures are unrelated (fail in other PRs as well)

**Pull request merged**